### ISSUE TITLE
There are some differences between the paper and the code

### ISSUE THREAD

**Issue by wxy1234567** [yuantn/MIAL#4]
In Figure 2 of the paper, first maximize instance uncertainty, then minimize instance uncertainty, but the code seems to be the opposite, and the frozen layers in different stages of the code are also the opposite

**Comment by yuantn** [comment]
> 
> 
> In Figure 2 of the paper, first maximize instance uncertainty, then minimize instance uncertainty, but the code seems to be the opposite, and the frozen layers in different stages of the code are also the opposite

Yes, we have noticed the problem. However, some of our previous experiments have shown that, if the order of max step and min step is reversed (including the frozen layers), the performance will change little. We are reproducing this now, and we will provide this log file soon. You can compare the log files in these two cases.

**Comment by yuantn** [comment]
> In Figure 2 of the paper, first maximize instance uncertainty, then minimize instance uncertainty, but the code seems to be the opposite, and the frozen layers in different stages of the code are also the opposite

And here are the results under the new condition, while the data in brackets is the last row in Tab. 1 in the paper:
**48.41** (47.18) | **56.16** (58.03) | **65.50** (63.98) | **68.65** (66.58) | **69.53** (69.57) | **70.88** (70.96) | **71.92** (72.03)
Here is the output log: [Google Drive](https://drive.google.com/file/d/1LcSmLbqaXTEr2zoe2ZUiVh8gCC2kfcHW/view?usp=sharing) | [Baidu Drive (Extraction Code: 3nza)](https://pan.baidu.com/s/1MvbG_i-KVVm9cXN1_tEj-A)

**Comment by wxy1234567** [comment]
Thank you ! There is also a small question. The initial labeled experiment in Figure 5 of this paper should be similar in theory. Is it because your algorithm needs to train more epochs?

**Comment by yuantn** [comment]
> Thank you ! There is also a small question. The initial labeled experiment in Figure 5 of this paper should be similar in theory. Is it because your algorithm needs to train more epochs?

No. As described in the subsection "Active Learning Settings" in Section 4.1, MIAL and other methods shared the same training setting, the same initialization and the same random seed.

The reason of performance improvement has been explained in the last paragraph in Section 3.2, as:

```In each active learning cycle, the max-min prediction discrepancy procedures repeat several times so that the instance uncertainty is learned and THE INSTANCE DISTRIBUTIONS of the labeled and unlabeled set ARE PROGRESSIVELY ALIGNED. This actually defines AN UNSUPERVISED LEARNING PROCEDURE, which leverages the information (i.e., PREDICTION DISCREPANCY) of the unlabeled set to improve the detection model.```

So the reason can be summarized as:
Intentional use of unlabeled data
-> Better aligned instance distributions of the labeled and unlabeled set
-> Effective information (prediction discrepancy) of the unlabeled set
-> Naturally formed unsupervised learning procedure
-> Performance improvement

And the internal direct module effect has been shown in Tab.1. Using 5.0% data, IUL can improve it from 28.31 to 30.09, and IUR can further improve it to 47.18.

**Comment by wxy1234567** [comment]
Thank you for your reply and solved my confusion !

**Comment by yuantn** [comment]
> 
> 
> Thank you for your reply and solved my confusion !

😄
